src/models/shopping-cart.model.ts

```diff
@@ -25,6 +25,10 @@ export class ShoppingCartItem {
    */
   @property()
   price?: number;
+
+  constructor(data?: Partial<ShoppingCartItem>) {
```

Comment:
Is the partial needed in this case since this isn't a model directly exposed over `REST` and won't face issues of trying to create an instance without a `productId`?

Reply:
Not a big deal as the `data` itself is optional. So we can create a empty instance and populate the data afterward. The `required` constraint should be enforced by model validation.

Reply:
> The required constraint should be enforced by model validation.

@raymondfeng Please note that KeyValue repository does not perform validation right now! See [lib/kvao/set.js](https://github.com/strongloop/loopback-datasource-juggler/blob/fcac276955a58c4aabcd2b602af046f45cfc73fe/lib/kvao/set.js). I think this is an oversight we should eventually fix.